onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.h

```diff
@@ -37,6 +39,17 @@ class MKLDNNExecutionProvider : public IExecutionProvider {
   }
 
   virtual std::shared_ptr<KernelRegistry> GetKernelRegistry() const override;
+
+  std::shared_ptr<mkldnn::memory> GetWeightMemory(std::string weightName) {
+    if (weights_mem_map.find(weightName) != weights_mem_map.end())
+      return weights_mem_map[weightName];
+    else
+      return nullptr;
+  }
+public:
+  // mkldnn formatted weights(filer data) memory from first iteration 
+  // saved by weights name
+  std::map<std::string, std::shared_ptr<mkldnn::memory>> weights_mem_map;
```

Comment:
the risk here, is that the key may be unique for a given model, but it may not be across models....

Reply:
better address this, else there may be strange failures when running multiple models.

Reply:
Yes. I did think about it. I can add weight dimensions to name as a key.

Reply:
adding weight dimensions may not be sufficient...

Reply:
thanks @jywu-msft. Will look at it. Do you have any suggestion to make it unique?

Reply:
@jywu-msft Let me know if string concatenation of the following will make the key unique.

graph name: `resnet50`
ConvParams Key:  `Key: "conv_#1_3_224_224_##64_3_7_7_####1_64_112_112_##2_2_##0_0_##3_3_##3_3_#`
weightId:  `gpu_0/conv1_w_0`

To get Graph name, can I add `GetGraphName()` to Node class in Graph.cc?

Thanks

Reply:
Based on the ONNX standard, model_version + graph name + domain forms the unique identity of the graph. See message ModelProto in onnx-ml.proto file.

  // Domain name of the model.
  // We use reverse domain names as name space indicators. For example:
  // `com.facebook.fair` or `com.microsoft.cognitiveservices`
  //
  // Together with `model_version` and GraphProto.name, this forms the unique identity of
  // the graph.
  optional string domain = 4;

Reply:
@pranavsharma @jywu-msft 
weight id by itself is sufficient. The provider instance (and its state) is unique to a session and we can only load a single model in a session. Therefore, weight id will be unique in the map.